libs/platform/src/lib/table-helpers/services/table-column-resize.service.ts

```diff
@@ -176,8 +176,15 @@ export class TableColumnResizeService implements OnDestroy {
 
     /** Retrieves custom column value or returns `unset` */
     getColumnWidthStyle(columnName: string): string {
-        const calculatedWidth = this._fixedColumnsWidthMap.get(columnName);
-        return calculatedWidth || 'unset';
+        if (this._tableRef._virtualScrollDirective?.scrollWholeRows) {
+            return (
+                this._fixedColumnsWidthMap.get(columnName) ||
+                this._tableRef._tableWidthPx / this._tableRef.getVisibleTableColumns().length + 'px'
+            );
```

Comment:
I think this `this._tableRef._tableWidthPx` needs optimization if we need this. I observe removing this new condition contributes performance improvement.

Reply:
This should only need to be called on init

Reply:
This is at `td` level. We basically do DOM query on potential reflow properties for every cell. And it is used as function call on table-row html (which is recommended to be avoided). May be converted to pipe also or avoid using DOM query here.